hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -576,6 +576,16 @@ public void shutdown(boolean terminate) {
         if (diagnostics != null) {
             diagnostics.shutdown();
         }
+
+        closeExternalDataStoreService();
+    }
+
+    private void closeExternalDataStoreService() {
+        try {
+            externalDataStoreService.close();
```

Comment:
WDYT of adding a timeout here? Reasoning is during non-graceful shutdown, externalDataStoreService can cause longer waiting times since it is non-hazelcast code, in current hikari code it seems it can wait max around 30 seconds.